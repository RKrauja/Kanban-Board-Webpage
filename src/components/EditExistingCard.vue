<script lang="ts" setup>
import { defineEmits, defineProps, ref } from "vue";

const props = defineProps({
  title: String,
  description: String,
});

const emits = defineEmits(["EditedTask", "editBlockVisibility"]);
const editedTitle = ref(props.title);
const editedDescription = ref(props.description);

const taskSaved = () => {
  emits("EditedTask", {
    title: editedTitle.value || "",
    description: editedDescription.value,
  });
  emits("editBlockVisibility", false);
};

const taskCancelled = () => {
  emits("EditedTask", {
    title: props.title,
    description: props.description,
  });
  emits("editBlockVisibility", false);
};
</script>

<template>
  <div class="pa-4">
    <v-text-field
      v-model="editedTitle"
      bg-color="white"
      class="mb-2"
      density="comfortable"
      hide-details
      placeholder="Enter a title..."
      variant="outlined"
    />

    <v-text-field
      v-model="editedDescription"
      bg-color="white"
      class="mb-3"
      density="compact"
      hide-details
      placeholder="Enter a description..."
      variant="outlined"
    />

    <div class="d-flex gap-2">
      <v-btn color="success" variant="flat" @click="taskSaved">SAVE</v-btn>
      <v-btn color="error" variant="text" @click="taskCancelled">CANCEL</v-btn>
    </div>
  </div>
</template>
