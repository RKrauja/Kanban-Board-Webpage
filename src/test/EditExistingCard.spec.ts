import { describe, expect, it } from "vitest";
import { mount } from "@vue/test-utils";
import { createVuetify } from "vuetify";
import EditExistingCard from "@/components/EditExistingCard.vue";

describe("AddNewTaskBlock", () => {
  const vuetify = createVuetify();
  const wrapper = mount(EditExistingCard, {
    global: {
      plugins: [vuetify],
    },
    props: {
      title: "test title",
      description: "test description",
    },
  });
  it("can render", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("can cancel edit of a task", async () => {
    const [titleField, descriptionField] = wrapper.findAll(
      ".v-text-field input",
    );
    await titleField.setValue("edited title");
    await descriptionField.setValue("edited description");
    const cancelButton = wrapper.findAll(".v-btn")[1];
    await cancelButton.trigger("click");
    expect(wrapper.emitted("EditedTask")?.[0][0]).toEqual({
      title: "test title",
      description: "test description",
    });
  });
});
