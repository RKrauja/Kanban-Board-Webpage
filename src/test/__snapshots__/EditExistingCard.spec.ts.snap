// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddNewTaskBlock > can render 1`] = `
"<div class="pa-4">
  <div class="v-input v-input--horizontal v-input--center-affix v-input--density-comfortable v-theme--light v-locale--is-ltr v-input--dirty v-text-field mb-2">
    <!---->
    <div class="v-input__control">
      <div class="v-field v-field--active v-field--center-affix v-field--dirty v-field--has-background v-field--no-label v-field--variant-outlined v-theme--light bg-white v-locale--is-ltr">
        <div class="v-field__overlay"></div>
        <div class="v-field__loader">
          <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
            <!---->
            <div class="v-progress-linear__background" style="opacity: NaN;"></div>
            <div class="v-progress-linear__buffer" style="opacity: NaN; width: 0%;"></div>
            <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
              <div class="v-progress-linear__indeterminate">
                <div class="v-progress-linear__indeterminate long"></div>
                <div class="v-progress-linear__indeterminate short"></div>
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <!---->
        <div class="v-field__field" data-no-activator="">
          <!----><label class="v-label v-field-label" for="input-0">
            <!---->
            <!---->
          </label>
          <!----><input placeholder="Enter a title..." size="1" type="text" id="input-0" aria-describedby="input-0-messages" class="v-field__input" value="test title">
          <!---->
        </div>
        <!---->
        <!---->
        <div class="v-field__outline">
          <div class="v-field__outline__start"></div>
          <!---->
          <div class="v-field__outline__end"></div>
          <!---->
        </div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
  <div class="v-input v-input--horizontal v-input--center-affix v-input--density-compact v-theme--light v-locale--is-ltr v-input--dirty v-text-field mb-3">
    <!---->
    <div class="v-input__control">
      <div class="v-field v-field--active v-field--center-affix v-field--dirty v-field--has-background v-field--no-label v-field--variant-outlined v-theme--light bg-white v-locale--is-ltr">
        <div class="v-field__overlay"></div>
        <div class="v-field__loader">
          <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
            <!---->
            <div class="v-progress-linear__background" style="opacity: NaN;"></div>
            <div class="v-progress-linear__buffer" style="opacity: NaN; width: 0%;"></div>
            <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
              <div class="v-progress-linear__indeterminate">
                <div class="v-progress-linear__indeterminate long"></div>
                <div class="v-progress-linear__indeterminate short"></div>
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <!---->
        <div class="v-field__field" data-no-activator="">
          <!----><label class="v-label v-field-label" for="input-2">
            <!---->
            <!---->
          </label>
          <!----><input placeholder="Enter a description..." size="1" type="text" id="input-2" aria-describedby="input-2-messages" class="v-field__input" value="test description">
          <!---->
        </div>
        <!---->
        <!---->
        <div class="v-field__outline">
          <div class="v-field__outline__start"></div>
          <!---->
          <div class="v-field__outline__end"></div>
          <!---->
        </div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
  <div class="d-flex gap-2"><button type="button" class="v-btn v-theme--light bg-success v-btn--density-default v-btn--size-default v-btn--variant-flat"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator="">SAVE</span>
      <!---->
      <!---->
    </button><button type="button" class="v-btn v-theme--light text-error v-btn--density-default v-btn--size-default v-btn--variant-text"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator="">CANCEL</span>
      <!---->
      <!---->
    </button></div>
</div>"
`;
